src/graph.py

```diff
@@ -0,0 +1,178 @@
+"""Graphy mcgraphface"""
```

Comment:
My PEP 8 linter is not happy with this entire file, you need to follow pep 8 styles. Docstrings are needed for all functions, and a space between function definitions!